### PR TITLE
Update to Java 16

### PR DESCRIPTION
**Pull request by modmuss50**

### CODE REVIEW

**Comment by modmuss50** [comment]
Im not quite sure about the test failiures, that will need looking into.

**Comment by YanisBft** [comment]
May be caused by inner class constructors byecode changes in enigma tests. I'm on phone tho, hard to see.

Otherwise changing the default decompiler is a good idea 👍

**Comment by liach** [comment]
I think this might be procyon not able to decompile modern java's constructor properly, as procyon is based on java 8 and the tests of enigma run with procyon

**Comment by Geolykt** [comment]
Isn't CFR used as the decompiler now?

**Comment by liach** [comment]
> 
> 
> Isn't CFR used as the decompiler now?

Ah, that might be it. I don't think cfr is exporting tokens for constructors as constructors but as classes? So the tests might break? Possible, but the actual cause always lies in the test result html.

**Comment by modmuss50** [comment]
Moving to CRF in the tests helped, some are still broken.

**Comment by liach** [comment]
Yeah, I didn't implement the entry references indexing within method bodies as cfr's dumper system ain't particularly suited for that.

File: build.gradle [hunk 1]
```diff
@@ -6,8 +6,8 @@ subprojects {
     apply plugin: 'java'
     apply plugin: 'maven-publish'
 
-    sourceCompatibility = 1.8
-    targetCompatibility = 1.8
+    sourceCompatibility = 16
+    targetCompatibility = 16
```
Comment (YanisBft) [inline build.gradle:10]:
```suggestion
    sourceCompatibility = JavaVersion.VERSION_16
    targetCompatibility = JavaVersion.VERSION_16
```
Better to use constants then literals

**Comment by liach** [comment]
@modmuss50 Do you mind temporarily marking those tests with `@Ignore` and indicate this is because of #368? I should find some time and work on fixing that

**Review by liach — APPROVED** [review]

File: enigma/build.gradle [hunk 2]
```diff
@@ -11,7 +11,7 @@ dependencies {
     implementation 'net.fabricmc:procyon-fabric-compilertools:0.5.35.13'
     implementation 'net.fabricmc:cfr:0.0.3'
 
-    proGuard 'com.guardsquare:proguard-base:7.1.0-beta1'
+    proGuard 'com.guardsquare:proguard-base:7.1.0-beta3'
```
Comment (liach) [inline enigma/build.gradle:14]: https://github.com/Guardsquare/proguard/commit/036291e8fb91d4aac26fc2ac2dbbe4d09ad5ae32 They are now on beta 5

**Comment by modmuss50** [comment]
I will ignore them for now 👍 Didnt really think of that.

File: enigma/build.gradle (hunk 2 above)
Comment (modmuss50) [reply enigma/build.gradle:14]: Seems to just be beta 4.

**Review by 2xsaiko — APPROVED** [review]

**Comment by modmuss50** [comment]
Thoughts on doing a 1.0 release before pulling this in? 

And then starting on 1.1 or 2.0? (As its fairly breaking)

**Comment by liach** [comment]
Hmm, Imo `0.` series would be for java 8 and `1.` version series would be for 16.

**Comment by modmuss50** [comment]
Thats resonable as well, mainly wanted to leave some space incase there is want to contiune with the j8 release line.

**Review by YanisBft — APPROVED** [review]
1.0 could be nice yeah

**Pull request merged**